exec/java-exec/src/main/java/org/apache/drill/exec/store/pcap/PcapRecordReader.java

```diff
@@ -0,0 +1,295 @@
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.drill.exec.store.pcap.Utils.parseBytesToASCII;
+
+public class PcapRecordReader extends AbstractRecordReader {
+
+  private OutputMutator output;
+
+  private final PacketDecoder decoder;
+  private ImmutableList<ProjectedColumnInfo> projectedCols;
+
+  private byte[] buffer = new byte[100000];
+  private int offset = 0;
+  private InputStream in;
+  private int validBytes;
+
+  private static final Map<PcapTypes, MinorType> TYPES;
+
+  private static class ProjectedColumnInfo {
+    ValueVector vv;
+    ColumnDto pcapColumn;
+  }
+
+  static {
+    TYPES = ImmutableMap.<PcapTypes, TypeProtos.MinorType>builder()
+        .put(PcapTypes.STRING, MinorType.VARCHAR)
+        .put(PcapTypes.INTEGER, MinorType.INT)
+        .put(PcapTypes.LONG, MinorType.BIGINT)
+        .put(PcapTypes.TIMESTAMP, MinorType.TIMESTAMP)
+        .build();
+  }
+
+  public PcapRecordReader(final String inputPath,
+                          final List<SchemaPath> projectedColumns) {
+    try {
+      this.in = new FileInputStream(inputPath);
+      this.decoder = getPacketDecoder();
+      validBytes = in.read(buffer);
+    } catch (IOException e) {
+      throw new RuntimeException("File " + inputPath + " not Found");
+    }
+    setColumns(projectedColumns);
+  }
+
+  @Override
+  public void setup(final OperatorContext context, final OutputMutator output) throws ExecutionSetupException {
+    this.output = output;
+  }
+
+  @Override
+  public int next() {
+    projectedCols = getProjectedColsIfItNull();
+    try {
+      return parsePcapFilesAndPutItToTable();
+    } catch (IOException io) {
+      throw new RuntimeException("Trouble with reading packets in file!");
+    }
+  }
+
+  @Override
+  public void close() throws Exception {
+  }
+
+  private PacketDecoder getPacketDecoder() {
+    try {
+      return new PacketDecoder(in);
+    } catch (IOException io) {
+      throw new RuntimeException("File Not Found or some I/O issue");
+    }
+  }
+
+  private ImmutableList<ProjectedColumnInfo> getProjectedColsIfItNull() {
+    return projectedCols != null ? projectedCols : initCols(new Schema());
+  }
+
+  private ImmutableList<ProjectedColumnInfo> initCols(final Schema schema) {
+    ImmutableList.Builder<ProjectedColumnInfo> pciBuilder = ImmutableList.builder();
+    ColumnDto column;
+
+    for (int i = 0; i < schema.getNumberOfColumns(); i++) {
+      column = schema.getColumnByIndex(i);
+
+      final String name = column.getColumnName();
+      final PcapTypes type = column.getColumnType();
+      TypeProtos.MinorType minorType = TYPES.get(type);
+
+      ProjectedColumnInfo pci = getProjectedColumnInfo(column, name, minorType);
+      pciBuilder.add(pci);
+    }
+    return pciBuilder.build();
+  }
+
+  private ProjectedColumnInfo getProjectedColumnInfo(final ColumnDto column,
```

Comment:
@paul-rogers 

> The code handle the second type: a list of columns. It does not seem to handle SELECT *, however.
Why you say that this case is not working ? Or I missing something ? Both cases you listed are operational